plugins/gcp/resoto_plugin_gcp/resources/compute.py

```diff
@@ -3292,7 +3296,92 @@ def collect_individual(cls: Type[GcpResource], builder: GraphBuilder, zone: str,
         machine_type_obj = GcpMachineType.from_api(result)
         builder.add_node(machine_type_obj, result)
 
-    # TODO post process
+    def connect_in_graph(self, builder: GraphBuilder, source: Json) -> None:
+        """Adds edges from machine type to SKUs and determines ondemand pricing
+        TODO: Implement GPU types
+        """
+
+        # log.debug((f"Looking up pricing for {self.rtdname}" f" in {self.location(graph).rtdname}"))
+        skus = []
+        for sku in builder.resources_of(GcpSku):
+            if not self.name or not self._region or not sku.name or not sku.category or not sku.geo_taxonomy:
+                continue
+
+            if not (sku.category.resource_family == "Compute" and sku.category.usage_type == "OnDemand"):
+                continue
+            if sku.category.resource_group not in (
+                "G1Small",
+                "F1Micro",
+                "N1Standard",  # ?
+                "CPU",
+                "RAM",
+            ):
+                continue
+            if ("custom" not in self.name and "Custom" in sku.name) or (
+                "custom" in self.name and "Custom" not in sku.name
+            ):
+                continue
+            if self._region.name not in sku.geo_taxonomy.regions:
+                continue
+            if self.name == "g1-small" and sku.category.resource_group != "G1Small":
+                continue
+            if self.name == "f1-micro" and sku.category.resource_group != "F1Micro":
+                continue
+            if (self.name.startswith("n2d-") and not sku.name.startswith("N2D AMD ")) or (
+                not self.name.startswith("n2d-") and sku.name.startswith("N2D AMD ")
+            ):
+                continue
+            if (self.name.startswith("n2-") and not sku.name.startswith("N2 ")) or (
+                not self.name.startswith("n2-") and sku.name.startswith("N2 ")
+            ):
+                continue
+            if (self.name.startswith("m1-") and not sku.name.startswith("Memory-optimized ")) or (
+                not self.name.startswith("m1-") and sku.name.startswith("Memory-optimized ")
+            ):
+                continue
+            if (self.name.startswith("c2-") and not sku.name.startswith("Compute optimized ")) or (
+                not self.name.startswith("c2-") and sku.name.startswith("Compute optimized ")
+            ):
+                continue
+            if self.name.startswith("n1-") and sku.category.resource_group != "N1Standard":
+                continue
+            if "custom" not in self.name:
+                if (self.name.startswith("e2-") and not sku.name.startswith("E2 ")) or (
+                    not self.name.startswith("e2-") and sku.name.startswith("E2 ")
+                ):
+                    continue
+            skus.append(sku)
```

Comment:
Looking up the available machine types - the following do not seem to be handled:
```
a2
c2
c2d
c3
m2
m3
t2a
t2d
```

Reply:
I stuck to what is being handled in the current collector for parity. Should I extend coverage instead?

Reply:
> Looking up the available machine types - the following do not seem to be handled:
> 
> ```
> a2
> c2
> c2d
> c3
> m2
> m3
> t2a
> t2d
> ```

I was unable to identify the relevant SKUs for M2 types